### PR TITLE
 Merge bitcoin#18584: test: Check that the version message does not leak the local address

### PR DESCRIPTION
**Pull request by vijaydasmp**
Backporting Merge bitcoin#18584: test: Check that the version message

### CODE REVIEW

**Review by UdjinM6 — APPROVED** [review]
utACK

File: test/functional/p2p_leak.py [hunk 1]
```diff
@@ -88,11 +92,24 @@ def on_version(self, message):
...
+
 class P2PLeakTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
         self.extra_args = [['-banscore=' + str(banscore)]]
 
+    def setup_network(self):
+        self.disable_mocktime()
+        self.setup_nodes()
+
```
Comment (PastaPastaPasta) [inline test/functional/p2p_leak.py:112]: Why is this needed? Are we missing another PR or something?

Comment (UdjinM6) [reply test/functional/p2p_leak.py:112]: That's because (unlike in btc) our tests use mocktime by default (so that blocks are generated starting with genesis time instead of current time). This test relies on `time` to get `time_begin` (and `time_end`) so `assert_greater_than_or_equal(ver.nTime, time_begin)` is going to fail without this patch.

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge
